src/state/translations-mixin.ts

```diff
@@ -180,12 +180,27 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
 
     private _applyTranslations(hass: HomeAssistant) {
       document.querySelector("html")!.setAttribute("lang", hass.language);
-      this.style.direction = computeRTL(hass) ? "rtl" : "ltr";
+      this._applyDirection(hass);
       this._loadCoreTranslations(hass.language);
       this.__loadedFragmetTranslations = new Set();
       this._loadFragmentTranslations(hass.language, hass.panelUrl);
     }
 
+    private _applyDirection(hass: HomeAssistant) {
+      if (computeRTL(hass)) {
+        this.style.direction = "rtl";
+        // apply custom properties used to fix RTL appearance throughout the system
+        this.style.setProperty("--rtl-12px", "12px");
+        this.style.setProperty("--rtl--8px", "-8px");
```

Comment:
I think this will get out of hand with variables, do we have an idea of how many variables we would need?

Also, would it not be better to not make specific `rtl` variables, but to also use those in `ltr` cases?

Reply:
As discussed - It wouldn't be too many since we just need end-values. We shouldn't do it for LTR since it would be less readable for most devs and LTR rules over RTL.